pallets/nfts/src/lib.rs

```diff
@@ -233,25 +269,49 @@ impl<T: Config> NFTs for Pallet<T> {
     type AccountId = T::AccountId;
     type NFTDetails = T::NFTDetails;
     type NFTId = T::NFTId;
+    type NFTSeriesId = T::NFTSeriesId;
 
     fn create(
         owner: &Self::AccountId,
         details: Self::NFTDetails,
+        series_id: Self::NFTSeriesId,
     ) -> result::Result<Self::NFTId, DispatchError> {
-        let id = Total::<T>::get();
-        Total::<T>::put(id.checked_add(&1.into()).ok_or(Error::<T>::NFTIdOverflow)?);
+        // Check if the owner is even allowed to add anything to the series.
+        if series_id != Default::default() {
+            let series_owner = Series::<T>::get(series_id).0;
+            ensure!(
+                series_owner == Default::default() || series_owner == *owner,
```

Comment:
Instead of the `Default::default` flow we probably should use an `Option` instead

Reply:
Done :)